docs/how-to/hip_runtime_api/asynchronous.rst

```diff
@@ -0,0 +1,487 @@
+.. meta::
+  :description: This topic describes asynchronous concurrent execution in HIP
+  :keywords: AMD, ROCm, HIP, asynchronous concurrent execution, asynchronous, async, concurrent, concurrency
+
+.. _asynchronous_how-to:
+
+*******************************************************************************
+Asynchronous concurrent execution
+*******************************************************************************
+
+Asynchronous concurrent execution important for efficient parallelism and
+resource utilization, with techniques such as overlapping computation and data
+transfer, managing concurrent kernel execution with streams on single or
+multiple devices or using HIP graphs.
+
+Streams and concurrent execution
+===============================================================================
+
+All asynchronous APIs, such as kernel execution, data movement and potentially
+data allocation/freeing all happen in the context of device streams.
+
+Streams are FIFO buffers of commands to execute in order on a given device.
+Commands which enqueue tasks on a stream all return promptly and the command is
+executed asynchronously. Multiple streams may point to the same device and
+those streams may be fed from multiple concurrent host-side threads. Execution
+on multiple streams may be concurrent but isn't required to be.
+
+Managing streams
+-------------------------------------------------------------------------------
+
+Streams enable the overlap of computation and data transfer, ensuring
+continuous GPU activity. 
+
+To create a stream, the following functions are used, each returning a handle
+to the newly created stream:
+
+- :cpp:func:`hipStreamCreate`: Creates a stream with default settings.
+- :cpp:func:`hipStreamCreateWithFlags`: Allows creating a stream, with specific
+  flags, listed below, enabling more control over stream behavior:
+
+  - ``hipStreamDefault``: creates a default stream suitable for most
+    operations. It ensures that the stream is not non-blocking.
```

Comment:
```suggestion
    operations. The default stream is a blocking operation.
```

Reply:
It seems like in this case, the only reason to use hipStreamCreateWithFlags is to create non-blocking streams?